File: DuDoong-Domain/src/main/java/band/gosrock/domain/domains/ticket_item/repository/TicketItemRepository.java
package band.gosrock.domain.domains.ticket_item.repository;


import band.gosrock.domain.domains.ticket_item.domain.TicketItem;
import band.gosrock.domain.domains.ticket_item.domain.TicketItemStatus;
import java.util.List;
import java.util.Optional;
import org.springframework.data.jpa.repository.JpaRepository;

public interface TicketItemRepository extends JpaRepository<TicketItem, Long> {

    List<TicketItem> findAllByEventIdAndTicketItemStatus(Long eventId, TicketItemStatus status);

    Boolean existsByEventId(Long eventId);

    Optional<TicketItem> findByIdAndTicketItemStatus(Long ticketItemId, TicketItemStatus status);
}

File: DuDoong-Domain/src/main/kotlin/band/gosrock/domain/domains/ticket_item/repository/TicketItemRepository.kt
package band.gosrock.domain.domains.ticket_item.repository

import band.gosrock.domain.domains.ticket_item.domain.TicketItem
import band.gosrock.domain.domains.ticket_item.domain.TicketItemStatus
import org.springframework.data.jpa.repository.JpaRepository
import java.util.Optional

interface TicketItemRepository : JpaRepository<TicketItem, Long> {

    fun findAllByEventIdAndTicketItemStatus(eventId: Long, status: TicketItemStatus): List<TicketItem>

    fun existsByEventId(eventId: Long): Boolean

    fun findByIdAndTicketItemStatus(ticketItemId: Long, status: TicketItemStatus): Optional<TicketItem>
}
